2022/Q3/artifacts/PoS-tla/MC_Staking.tla

```diff
@@ -160,34 +171,136 @@ HighCoverage(trace) ==
 
 \* PoS invariants
 
-\* From Chris
-\* Try to capture that for a group of validators with total voting power X at
-\* a particular height, the proof-of-stake model provides the ability to look up
-\* bonds contributing to that voting power (for the unbonding period) and slash
-\* a proportional amount of stake to X (subject to limitations e.g. repeated infractions)
+(* 
+ * From Chris:
+ * Try to capture that for a group of validators with total voting power X at
+ * a particular height, the proof-of-stake model provides the ability to look up
+ * bonds contributing to that voting power (for the unbonding period) and slash
+ * a proportional amount of stake to X (subject to limitations e.g. repeated infractions)
+ *
+ * We can break this into two properties:
+ *  (i) Given a validators with total voting power X at a particular height, the 
+ *      proof-of-stake model provides the ability to look up bonds contributing to
+ *      that voting power.
+ * (ii) If evidence is found in time and there are no repeated infractions within 
+ *      an epoch window, then the proof-of-stake model guarantees slashing.
+ *
+ * Invariant #6 verifies property (i) for the case in which there is no slashing.
```

Comment:
I am not sure I am following. With slashing, there will be no match between total_deltas and bonds (bonds are not updated when slashing). Invariant 4 is similar to this one but with slashing, but in that case sum(bonds) >= total_deltas because of the above reason.